vpr/src/place/move_transactions.cpp

```diff
@@ -2,6 +2,18 @@
 
 #include "globals.h"
 #include "place_util.h"
+#include "vtr_assert.h"
+
+t_pl_blocks_to_be_moved::t_pl_blocks_to_be_moved(size_t max_blocks){
+        moved_blocks.reserve(max_blocks);
+        moved_blocks.resize(0);
+}
+
+size_t t_pl_blocks_to_be_moved::get_size_and_increment() {
+    VTR_ASSERT_SAFE(moved_blocks.size() < moved_blocks.capacity());
+    moved_blocks.resize(moved_blocks.size() + 1);
+    return moved_blocks.size() - 1;
+}
 
 //Records that block 'blk' should be moved to the specified 'to' location
 e_block_move_result record_block_move(t_pl_blocks_to_be_moved& blocks_affected, ClusterBlockId blk, t_pl_loc to) {
```

Comment:
this function does not access or modify the global state and only changes the content of blocks_affected. It would be more readable if this functon is implemeneted as a method under t_pl_blocks_to_be_moved.

Reply:
res.second and res2.second are not clear enough in this code. Now that you have touched this function, can you please make them more readable by using the following syntax?

```
auto [it, success] = blocks_affected.moved_to.emplace(to);

if (!success) {
}
```

Reply:
I guess there is a subtle bug in this function. If the move is aborted because `res.second `is false, we need to remove `to` from `blocks_affected.moved_to`.

Reply:
> I guess there is a subtle bug in this function. If the move is aborted because `res.second `is false, we need to remove `to` from `blocks_affected.moved_to`.

Should this apply to res2?

Reply:
> > I guess there is a subtle bug in this function. If the move is aborted because `res.second `is false, we need to remove `to` from `blocks_affected.moved_to`.
> 
> Should this apply to res2?

Yes. When we abort the move because `res.second is false`, `to` has already been inserted into `locks_affected.moved_to`.

Reply:
> > > I guess there is a subtle bug in this function. If the move is aborted because `res.second `is false, we need to remove `to` from `blocks_affected.moved_to`.
> > 
> > 
> > Should this apply to res2?
> 
> Yes. When we abort the move because `res.second is false`, `to` has already been inserted into `locks_affected.moved_to`.

Would this also apply to `res2` and `from`?

Reply:
> I guess there is a subtle bug in this function. If the move is aborted because `res.second `is false, we need to remove `to` from `blocks_affected.moved_to`.

I made a mistake here. I meant `res2.second`.
If `res.second` is false, `to` has not been inserted to the set.